src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtAddressService.java

```diff
@@ -68,11 +77,19 @@ public AdminCourtAddressService(final CourtRepository courtRepository, final Cou
         final List<uk.gov.hmcts.dts.fact.entity.CourtAddress> newCourtAddressesEntity = constructCourtAddressesEntity(courtEntity, courtAddresses);
         courtAddressRepository.deleteAll(courtEntity.getAddresses());
 
-        return courtAddressRepository.saveAll(newCourtAddressesEntity)
+        List<CourtAddress> updatedAddresses = courtAddressRepository.saveAll(newCourtAddressesEntity)
             .stream()
             .sorted(Comparator.comparingInt(a -> AddressType.isCourtAddress(a.getAddressType().getName()) ? 0 : 1))
             .map(CourtAddress::new)
             .collect(toList());
+        adminAuditService.saveAudit(
+            AuditType.findByName("Update court addresses and coordinates"),
```

Comment:
I'm trying to think of a way to avoid having to do a findByName call with a string for every audit, which might be error prone. Introduce an enum? Pass the audited class type and let the service contain the findByName logic?

Reply:
Yeah so AuditType is a class that checks based on a string that it is within a list of enums. So it isn't doing a sql call here. It checks it exists, and then uses it later on when it adds an audit to the db